src/main/scala/org/constellation/snapshotstreaming/TessellationServices.scala

```diff
@@ -3,46 +3,67 @@ package org.constellation.snapshotstreaming
-import org.tessellation.node.shared.config.types.SnapshotSizeConfig
 import org.tessellation.node.shared.infrastructure.block.processing.BlockAcceptanceManager
 import org.tessellation.node.shared.infrastructure.snapshot._
 import org.tessellation.node.shared.modules.SharedValidators
 import org.tessellation.schema.balance.Amount
 import org.tessellation.security.signature.SignedValidator
-import org.tessellation.security.{HashSelect, Hasher, SecurityProvider}
-
+import org.tessellation.security.Hasher
+import org.tessellation.security.HasherSelector
+import org.tessellation.security.SecurityProvider
 import eu.timepit.refined.auto._
+import org.tessellation.node.shared.domain.statechannel.FeeCalculator
 
 object TessellationServices {
 
-  def make[F[_]: Async: JsonSerializer: KryoSerializer: SecurityProvider: Hasher](configuration: Configuration, hashSelect: HashSelect, snapshotSizeConfig: SnapshotSizeConfig): F[TessellationServices[F]] =
+  def make[F[_]: Async: JsonSerializer: KryoSerializer: SecurityProvider](configuration: Configuration, hasherSelector: HasherSelector[F]): F[TessellationServices[F]] =
     for {
       _ <- Async[F].unit
-      validators = SharedValidators.make[F](None, None, None, snapshotSizeConfig.maxStateChannelSnapshotBinarySizeInBytes)
+      txHasher = Hasher.forKryo
+      validators = SharedValidators.make[F](None, None, None, configuration.feeConfigs, configuration.snapshotSize.maxStateChannelSnapshotBinarySizeInBytes, txHasher)
       currencySnapshotAcceptanceManager = CurrencySnapshotAcceptanceManager.make(
-        BlockAcceptanceManager.make[F](validators.currencyBlockValidator),
+        BlockAcceptanceManager.make[F](validators.currencyBlockValidator, txHasher),
         Amount(0L),
-        hashSelect
+        validators.currencyMessageValidator
       )
       currencyEventsCutter = CurrencyEventsCutter.make[F](None)
-      currencySnapshotCreator = CurrencySnapshotCreator.make[F](currencySnapshotAcceptanceManager, None, snapshotSizeConfig, currencyEventsCutter)
-      currencySnapshotValidator = CurrencySnapshotValidator.make[F](currencySnapshotCreator, SignedValidator.make[F], None, None)
-      currencySnapshotContextFns = CurrencySnapshotContextFunctions.make(currencySnapshotValidator, hashSelect)
+      alwaysCurrentHasherSelector = HasherSelector.forSyncAlwaysCurrent(Hasher.forJson)
+      currencySnapshotCreator = {
+        implicit val hs = alwaysCurrentHasherSelector
+        CurrencySnapshotCreator.make[F](currencySnapshotAcceptanceManager, None, configuration.snapshotSize, currencyEventsCutter)
+      }
+      currencySnapshotValidator = {
+        implicit val hs = alwaysCurrentHasherSelector
+        CurrencySnapshotValidator.make[F](currencySnapshotCreator, SignedValidator.make[F], None, None)
+      }
+      currencySnapshotContextFns = {
+        implicit val hs = alwaysCurrentHasherSelector
+        CurrencySnapshotContextFunctions.make(currencySnapshotValidator)
+      }
```

Comment:
Couldn't we make `alwaysCurrentHasherSelector` implicit? Then we don't have to create the implicit 3 times in blocks

Reply:
Or, if that's going to conflict with `hashSelector` later, use one block to create all three variables?

Reply:
I just tried the above and maybe doesn't read as well. WDYT?

Reply:
We only care about creating `currencySnapshotContextFns` so I am fine with wrapping in a single block :+1:

Reply:
Oh, actually we can even get rid of implicit current hasher selector for currency completely because currency uses Json only so there is no need to select hasher based on ordinal.